eng/Generate.ps1

```diff
@@ -310,6 +299,26 @@ if (![string]::IsNullOrWhiteSpace($filter))
     }
 }
 
+if ($reset -or $env:TF_BUILD)
+{
+    $cadlCount = ([string]::IsNullOrWhiteSpace($filter) ? $cadlDefinitions : $cadlDefinitions.Keys.Where({$_ -match $filter})).Count
+    $swaggerCount = $keys.Count - $cadlCount
+    if ($swaggerCount -gt 0) 
+    {
+        AutoRest-Reset;
+    }
+    
+    if ($cadlCount -gt 0) 
+    {
+        Invoke-CadlSetup
+    }
+}
+
+if (!$noBuild)
+{
+    Invoke "dotnet build $autoRestPluginProject"
+}
```

Comment:
Is there a flag which makes us to "include the build of the generator, but skip the CadlSetup"?
I believe in some scenarios, when we do not touch anything in the emitter, it does not have to rebuild everytime? Especially when we are changing something in the generator

Reply:
Sorry, I don't understand the question.
`Invoke-CadlSetup` builds emitter. By default (without any flags), cadl will be regenerated (similar to codemodel.yaml), but emitter itself won't be rebuilt.

Reply:
There is a separate PR for this change (https://github.com/Azure/autorest.csharp/pull/2614), I'm just working on top of it cause my machine is too slow, and I can't work without this fix.

Reply:
Thank you! I am actually asking if we have a flag that could build the generator and regenerate the test project but skip the cadl setup because it takes a long time and mgmt generator never needs it.
I suppose when working on DPG generator, not every time when you change something in the generator, you will have to involve `cadl setup` in the generate script

Reply:
> I am actually asking if we have a flag that could build the generator and regenerate the test project but skip the cadl setup

With this change, that would be a default behavior. To rebuild emitter you'll have to add `-reset` flag.